### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mutugi-c**
In this JavaScript capstone project milestone, a web application based on external APIs was built. The following steps were taken to complete this task:

1. API Integration:

- We used the pokeAPI, which provides the required data for the web application.
- We Implemented API requests to: 
   - Get a list of items with a unique item id.
   - Fetch detailed information about a given item.

2. User Interfaces:

- We created the home page interface and implemented the comments popup interface.

3. Data Retrieval and Display:

- We integrated the selected API to fetch data and display a list of items on the home page.
- We used the Involvement API to show item likes and comments on the respective interfaces.

4. User Interactions:

- Using the Involvement API, we implemented functionality to record user likes, comments, and reservations.
- The screen updates dynamically when users interact with items (e.g., Like button, Comment button).

5. UI Design:

- We Followed the provided wireframes to create the layout of the interfaces.
- We personalised the design by customising colours, typographies, and spacings.

6. Counters:

- We added counters to display the number of items, comments, and reservations in their interfaces.
- We Implemented specific functions to calculate these numbers based on the data received from the API.
- We Covered the count functions with unit tests using Jest.

7. Technical Setup:

- We set up the repository on GitHub and followed the Gitflow workflow.
- We configured webpack to bundle the JavaScript code.
- We also Integrated Jest as the JavaScript testing library for unit testing.

8. Project Documentation:

-We created a video presentation showcasing the project's features.
- The video includes a description of the project, a demonstration of its functionalities, and highlights of notable code segments.

### CODE REVIEW

File: src/tests/index.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    // Act
+    await commentsCounter(count);
+
+    // Assert
+    const countText = document.getElementById('comments-counter');
+    expect(countText.innerText).toBe(`Comments (${count})`);
+  });
+});
+
```
Comment (Zuhaib042) [inline src/tests/index.test.js:30]: - [ ] Kindly consider writing tests for edge cases for both pokemonsCount and commentsCounter.
think about the edge case when there is 0 pokemon, so consider adding a test when pokemonsCount is 0.
and also consider adding same test for commentsCounter where there is 0 comments.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @mutugi-c @SergioPeralta22 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Implemented features successfully
✅ Professional Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:✔️✔️ 

Hi @Team   ✔️✔️ ✔️✔️ 

Your project is complete! Great job for taking time to make the changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️✔️ 
- Test ok ✔️✔️ 
- design ok✔️✔️ 
- likes and comments✔️✔️ 
- No linter errors✔️✔️ 



---[optional] great job with the PR summary. Please always ensure to have a good PR title/description, so it can portray what you worked at from a glance.

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

smile

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
